ktor-client/ktor-client-core/jvm/src/io/ktor/client/features/HttpTimeoutJVM.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.features
+
+import io.ktor.client.request.*
+import io.ktor.util.*
+import io.ktor.utils.io.*
+import kotlinx.coroutines.*
+import java.net.*
+
+@Suppress("ACTUAL_WITHOUT_EXPECT")
+actual class HttpConnectTimeoutException actual constructor(request: HttpRequestData) :
+    ConnectException(
+        "Connect timeout has been expired [url=${request.url}, connect_timeout=${request.getCapabilityOrNull(
+            HttpTimeout
+        )!!.connectTimeoutMillis} ms]"
```

Comment:
So we get a KNPE exception when a connect exception is caught that is matching our timeout criteria but no timeout configured. I would at least put a placeholder here instead

Reply:
Done, I added the placeholder.